pkg/p2p/libp2p/libp2p.go

```diff
@@ -348,6 +347,25 @@ func buildUnderlayAddress(addr ma.Multiaddr, peerID libp2ppeer.ID) (ma.Multiaddr
 	return addr.Encapsulate(hostAddr), nil
 }
 
+func (s *Service) ConnectNotify(ctx context.Context, addr ma.Multiaddr) (address *bzz.Address, err error) {
+	info, err := libp2ppeer.AddrInfoFromP2pAddr(addr)
+	if err != nil {
+		return nil, fmt.Errorf("addr from p2p: %w", err)
+	}
+
+	address, err = s.Connect(ctx, addr)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
we can save a line, but i am trying to avoid writing these blocks because it might accidentally rescope variables inside the `if` block and i think this can cause problems at certain points. example:
```
var a
if a := abcd(); a != nil {
  // shadows a to exist only inside the if statement
}
// a is nil again
```
it is just enough to make a mistake and write `:=` instead of `=` and it might create some confusion and is harder to read IMO...

Reply:
fair enough